main.py

```diff
@@ -0,0 +1,382 @@
+        ab = AnnotationBbox(imagebox, (0.05, 0.01), frameon=False, xycoords="axes fraction", box_alignment=(0.5, 0.5))
+        ax.add_artist(ab)
+
+    def _add_small_sentence(self, ax: plt.Axes, font_size: float):
+        """
+        Add the small informational sentence to the ballot.
+
+        Parameters:
+            ax (plt.Axes): The axes to draw on.
+            font_size (float): The base font size to use.
+        """
+        ax.text(0.5, 0.02, SMALL_SENTENCE, fontsize=font_size * 0.75, ha="center", va="top")
+
+    def _add_table(self, ax: plt.Axes, font_size: float, candidates: List[str] = None):
+        """
+        Add the main table to the ballot.
+
+        Parameters:
+            ax (plt.Axes): The axes to draw on.
+            font_size (float): The font size to use for the table.
+            candidates (List[str], optional): List of candidate names. Defaults to None.
+        """
+        candidates = self.config.candidates.copy() if candidates is None else candidates
+
+        data = [["" for _ in self.config.grades] for _ in candidates]
+        table = ax.table(
+            cellText=data,
+            rowLabels=candidates,
+            colLabels=self.config.grades,
+            cellLoc="center",
+            loc="center",
+            bbox=[0.05, 0.05, 0.95, 0.85],
+        )
+        self._format_table(table, font_size)
+
+    def _format_table(self, table: plt.table, font_size: float):
+        """
+        Format the table on the ballot.
+
+        Parameters:
+            table (plt.Table): The table to format.
+            font_size (float): The font size to use.
+        """
+        table.auto_set_font_size(False)
+        table.set_fontsize(font_size)
+        table.scale(1, 1.5)
+
+        for j, grade in enumerate(self.config.grades):
+            table[(0, j)].set_facecolor(self.grade_colors[j])
+            table[(0, j)].set_text_props(color="white", weight="bold")
+
+        for cell in table._cells:
+            table._cells[cell].set_edgecolor("black")
+            table._cells[cell].set_linewidth(0.5)
+
+    def _save_ballot(self, fig: plt.Figure, output_filename: str):
+        """
+        Save the ballot as a PDF file.
+
+        Parameters:
+            fig (plt.Figure): The figure to save.
+            output_filename (str): Name of the output PDF file.
+        """
+        plt.tight_layout()
+        with PdfPages(output_filename) as pdf:
+            pdf.savefig(fig, bbox_inches="tight", orientation="landscape", dpi=600)
+        plt.close(fig)
+
+    def _generate_grade_colors(self) -> List[str]:
+        """
+        Generate a list of colors for the grades.
+
+        Returns:
+            List[str]: List of color codes for each grade.
+        """
+        num_grades = len(self.config.grades)
+
+        if num_grades > len(BASE_COLORS):
+            raise ValueError("More grades than available colors")
```

Comment:
Note for later: You could make a lerp between color RGB (11111111 11111111 11111111) (use mask to get octet of each component, then lerp each component and compose color)